src/Dapr.Actors/Runtime/Actor.cs

```diff
@@ -328,9 +291,46 @@ protected async Task UnregisterTimerAsync(IActorTimer timer)
         protected async Task UnregisterTimerAsync(string timerName)
         {
             await ActorRuntime.DaprInteractor.UnregisterTimerAsync(this.actorTypeName, this.Id.ToString(), timerName);
-            if (this.timers.ContainsKey(timerName))
+        }
+
+        internal MethodInfo GetMethodInfoUsingReflection(Type actorType, string callback)
+        {
+            return actorType.GetMethod(callback, BindingFlags.Public | BindingFlags.Instance | BindingFlags.NonPublic | BindingFlags.Static | BindingFlags.InvokeMethod);
+        }
+
+        internal void ValidateTimerCallback(ActorHost host, string callback)
+        {
+            var actorTypeName = host.ActorTypeInfo.ActorTypeName;
+            var actorType = host.ActorTypeInfo.ImplementationType;
+
+            MethodInfo methodInfo = default;
```

Comment:
nit: you don't need the initializer here (`MethodInfo methodInfo;` will do)

The compiler will verify that the variable is initialized before its accessed.